test/pkg/pods/pods.go

```diff
@@ -213,29 +213,71 @@ func DefinePodOnNode(namespace string, nodeName string) *corev1.Pod {
 	return pod
 }
 
-// RedefineAsPrivileged updates the pod definition to be privileged
-func RedefineAsPrivileged(pod *corev1.Pod, containerName string) (*corev1.Pod, error) {
-	c := containerByName(pod, containerName)
-	if c == nil {
-		return pod, fmt.Errorf("container with name: %s not found in pod", containerName)
-	}
-	if c.SecurityContext == nil {
-		c.SecurityContext = &corev1.SecurityContext{}
+// DefinePodOnHost creates the pod defintion with a mount to host
+func DefinePodOnHost(namespace string) (*corev1.Pod, error) {
+	pod := getDefinition(namespace)
+
+	pod = RedefineWithMount(pod, pod.Spec.Containers[0].Name,
+		corev1.Volume{
+			Name: "host",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/",
+				},
+			}},
+		corev1.VolumeMount{
+			Name:      "host",
+			MountPath: "/host"})
+
+	pod, err := RedefineAsPrivileged(pod, pod.Spec.Containers[0].Name)
+	if err != nil {
+		return nil, err
 	}
-	c.SecurityContext.Privileged = pointer.BoolPtr(true)
 
 	return pod, nil
 }
-func containerByName(pod *corev1.Pod, containerName string) *corev1.Container {
-	if containerName == "" {
-		return &pod.Spec.Containers[0]
+
+// RedefineWithMount updates the container definition with a volume and volume mount
+func RedefineWithMount(pod *corev1.Pod, containerName string, volume corev1.Volume, mount corev1.VolumeMount) *corev1.Pod {
+	for i, c := range pod.Spec.Containers {
+		if c.Name == containerName {
+			pod.Spec.Containers[i].VolumeMounts = []corev1.VolumeMount{mount}
+			pod.Spec.Volumes = []corev1.Volume{volume}
+		}
 	}
 
-	for _, c := range pod.Spec.Containers {
+	return pod
+}
+
+// RedefineAsPrivileged updates the container definition to be privileged
+func RedefineAsPrivileged(pod *corev1.Pod, containerName string) (*corev1.Pod, error) {
+	b := true
+	for i, c := range pod.Spec.Containers {
 		if c.Name == containerName {
-			return &c
+			pod.Spec.Containers[i].SecurityContext = &corev1.SecurityContext{
+				Privileged: &b,
+			}
 		}
 	}
 
-	return nil
+	return pod, nil
+}
+
+func GetPTPDaemonPods(api *client.ClientSet) ([]*corev1.Pod, error) {
```

Comment:
You are right, there are other items in the list only when running in a multi node cluster, not an SNO. Since these tests are for SNOs I think we can the original function to always return the first element and avoid having the helper. What do you think?

Reply:
For the purposes of the interface it feels more complete to have both, but you are right: Since we are targeting SNO there's only ever one pod. I'm happy to go with making it clear in the docstring that SNO is a precondition and add to the original function.